server/handler/status.go

```diff
@@ -50,44 +64,83 @@ func (h *Status) Handle(ctx context.Context, eventType, deliveryID string, paylo
-		logger.Debug().Msgf("Ignoring context event for '%s'", event.GetContext())
+	if sender.GetLogin() == h.PullOpts.AppName+"[bot]" {
 		return nil
 	}
 
-	sender := event.GetSender()
-	commitSHA := event.GetCommit().GetSHA()
+	logger.Warn().
+		Str(LogKeyAudit, event.GetName()).
+		Str(LogKeyGitHubSHA, commitSHA).
+		Msgf(
+			"Entity '%s' overwrote status check '%s' to state='%s' description='%s' targetURL='%s'",
+			sender.GetLogin(),
+			event.GetContext(),
+			event.GetState(),
+			event.GetDescription(),
+			event.GetTargetURL(),
+		)
+
+	// must be less than 140 characters to satisfy GitHub API
+	desc := fmt.Sprintf("'%s' overwrote status to '%s'", sender.GetLogin(), event.GetState())
+
+	// unlike in other code, use a single context here because we want to
+	// replace a forged context with a failure, not post a general status
+	// if multiple contexts are forged, we will handle multiple events
+	status := &github.RepoStatus{
+		Context:     event.Context,
+		State:       github.String("failure"),
+		Description: &desc,
+	}
 
-	if sender.GetLogin() != h.PullOpts.AppName+"[bot]" {
-		logger.Warn().
-			Str(LogKeyAudit, eventType).
-			Str(LogKeyGitHubSHA, commitSHA).
-			Msgf(
-				"Entity '%s' overwrote status check '%s' to state='%s' description='%s' targetURL='%s'",
-				sender.GetLogin(),
-				event.GetContext(),
-				event.GetState(),
-				event.GetDescription(),
-				event.GetTargetURL(),
-			)
-
-		// must be less than 140 characters to satisfy GitHub API
-		desc := fmt.Sprintf("'%s' overwrote status to '%s'", sender.GetLogin(), event.GetState())
-
-		// unlike in other code, use a single context here because we want to
-		// replace a forged context with a failure, not post a general status
-		// if multiple contexts are forged, we will handle multiple events
-		status := &github.RepoStatus{
-			Context:     event.Context,
-			State:       github.String("failure"),
-			Description: &desc,
-		}
+	_, _, err = client.Repositories.CreateStatus(ctx, ownerName, repoName, commitSHA, status)
+	return err
+}
 
-		_, _, err := client.Repositories.CreateStatus(ctx, ownerName, repoName, commitSHA, status)
+func (h *Status) processOthers(ctx context.Context, event github.StatusEvent) error {
+	repo := event.GetRepo()
+	ownerName := repo.GetOwner().GetLogin()
+	repoName := repo.GetName()
+	commitSHA := event.GetCommit().GetSHA()
+	installationID := githubapp.GetInstallationIDFromEvent(&event)
+
+	client, err := h.NewInstallationClient(installationID)
+	if err != nil {
 		return err
 	}
 
-	return nil
+	ctx, logger := githubapp.PrepareRepoContext(ctx, installationID, repo)
+
+	// In practice, there should only be one or two open PRs for a given commit. In exceptional cases, if there are
+	// more than 100 open PRs, only process the most recent 100.
+	prs, _, err := client.PullRequests.List(
+		ctx,
+		ownerName,
+		repoName,
+		&github.PullRequestListOptions{
+			Head:        commitSHA,
+			ListOptions: github.ListOptions{PerPage: 100},
```

Comment:
Unfortunately, I don't think this works. The `head` filter is documented as

> Filter pulls by head user or head organization and branch name in the format of `user:ref-name` or `organization:ref-name`.

not by the commit SHA. Assuming that's correct and it doesn't also accept SHA values, I think that leaves two options:

1. List _all_ open pull requests (with paging) until we find ones containing the SHA
2. Use the GraphQL API instead so we can use the `associatedPullRequests` connection on the [`commit`](https://developer.github.com/v4/object/commit/) type. 

I think my preference is for the GraphQL approach, since it should only require a single API call, although it doesn't support filtering by state, which we'll have to do client-side.

Reply:
There's a `ListPullRequestsWithCommit`, which seems like what we want, so switched to use that